### PR TITLE
Refactor: Enhance validation and formatting

### PR DESCRIPTION
**Pull request by MuhammedMagdyy**
##### Why do we need this change?
This PR is just for enhancing validation and formatting files.
- Added new validation like (toLowerCase, and nonempty).
- Formatted files.

### CODE REVIEW

File: src/modules/Post/services/postService.ts [hunk 1]
```diff
@@ -7,14 +7,14 @@ export class PostService {
...
     .required();
 
   updatePostSchema = z.object({
-    description: z.string().nonempty({
-      message: 'Description cannot be empty',
-    }),
+    description: z
+      .string()
+      .nonempty({ message: 'Description cannot be empty' }),
   });
```
Comment (kerolloz) [inline src/modules/Post/services/postService.ts:19]: I only update the description in the post? I can't update the image?

File: src/modules/User/services/userService.ts [hunk 2]
```diff
@@ -22,19 +22,26 @@ export class UserService {
         .min(3, { message: 'First name is too short' })
         .trim(),
       lastName: z.string().min(3, { message: 'Last name is too short' }).trim(),
-      email: z.string().email({ message: 'Invalid email' }).trim(),
+      email: z
+        .string()
+        .email({ message: 'Invalid email' })
+        .toLowerCase()
+        .trim(),
```
Comment (kerolloz) [inline src/modules/User/services/userService.ts:29]: where's the endsWith instead of refine (the check 'klivvr.com')

Comment (MuhammedMagdyy) [reply src/modules/User/services/userService.ts:29]: Should we add it now?
I thought we will add it in the last of our project

File: src/modules/ResetPasswordCode/services/resetPasswordCodeService.ts [hunk 3]
```diff
@@ -17,13 +17,11 @@ import { z } from 'zod';
...
   resetPasswordRequestSchema = z
-    .object({
-      email: z.string().email().trim(),
-    })
+    .object({ email: z.string().email().toLowerCase().trim() })
     .required();
   resetPasswordSchema = z
     .object({
-      email: z.string().email().trim(),
+      email: z.string().email().toLowerCase().trim(),
```
[the remark this answers is not in the document]
Comment (MuhammedMagdyy) [reply src/modules/ResetPasswordCode/services/resetPasswordCodeService.ts:24]: Thanks ❤

File: src/modules/Post/services/postService.ts (hunk 1 above)
Comment (kerolloz) [reply src/modules/Post/services/postService.ts:19]: also let the description be optional on the update schema

File: src/modules/User/services/userService.ts (hunk 2 above)
Comment (kerolloz) [reply src/modules/User/services/userService.ts:29]: hmmm, okay add a `// TODO: ` comment

File: src/modules/Post/services/postService.ts (hunk 1 above)
Comment (MuhammedMagdyy) [reply src/modules/Post/services/postService.ts:19]: OK

**Review by kerolloz — APPROVED** [review]

**Review by mohamadelhinamy — APPROVED** [review]

**Pull request merged**